java/monkey/src/main/java/root/ast/Statement.java

```diff
@@ -0,0 +1,7 @@
+package root.ast;
+
+public abstract class Statement extends Node {
+    public void statementNode() {
```

Comment:
Prefer this to be `public abstract void statementNode();` so that implementations are forced to define it, instead of an empty call.

Reply:
I kept both functions with default empty implementations for now because in the book they haven't been implemented yet, all all implementations would just be an empty function, but I do agree on changing this when the implementation is actually needed.